driver/configuration/provider_viper.go

```diff
@@ -153,6 +153,15 @@ func (p *ViperProvider) DSN() string {
 	return ""
 }
 
+func (p *ViperProvider) DSNAddress() string {
+	dsn := p.DSN()
+	if dsn == "memory" {
+		return "sqlite://:memory:?_fk=true"
```

Comment:
```suggestion
		return "sqlite://mem.db?mode=memory?_fk=true&cache=shared"
```

https://github.com/mattn/go-sqlite3/issues/204, https://github.com/mattn/go-sqlite3/issues/511, https://www.sqlite.org/sharedcache.html#shared_cache_and_in_memory_databases